File: modules/slf4j/src/main/scala/io/taig/flog/slf4j/StaticLoggerBinder.scala
package org.slf4j.impl

import io.taig.flog.slf4j.FlogSlf4jBinder
import org.slf4j.ILoggerFactory
import org.slf4j.helpers.NOPLoggerFactory
import org.slf4j.spi.LoggerFactoryBinder

abstract class StaticLoggerBinder extends LoggerFactoryBinder

object StaticLoggerBinder extends StaticLoggerBinder {
  final override def getLoggerFactory: ILoggerFactory =
    FlogSlf4jBinder.getFactory().getOrElse(new NOPLoggerFactory)

  final override val getLoggerFactoryClassStr: String = getClass.getName

  var REQUESTED_API_VERSION: String = "1.7.30"

  def getSingleton: StaticLoggerBinder = this
}

File: modules/slf4j/src/main/scala/org/slf4j/impl/StaticLoggerBinder.java
package org.slf4j.impl;

import io.taig.flog.slf4j.Build$;
import org.slf4j.ILoggerFactory;
import org.slf4j.spi.LoggerFactoryBinder;

public class StaticLoggerBinder implements LoggerFactoryBinder {
    /**
     * The unique instance of this class.
     */
    private static final StaticLoggerBinder SINGLETON = new StaticLoggerBinder();

    /**
     * Return the singleton of this class.
     *
     * @return the StaticLoggerBinder singleton
     */
    public static final StaticLoggerBinder getSingleton() {
        return SINGLETON;
    }

    /**
     * Declare the version of the SLF4J API this implementation is compiled against.
     * The value of this field is modified with each major release.
     */
    // to avoid constant folding by the compiler, this field must *not* be final
    public static String REQUESTED_API_VERSION = Build$.MODULE$.slf4jVersion(); // !final

    private static final String loggerFactoryClassStr = FlogLoggerFactory.class.getName();

    /**
     * The ILoggerFactory instance returned by the {@link #getLoggerFactory}
     * method should always be the same object
     */
    private final ILoggerFactory loggerFactory;

    private StaticLoggerBinder() {
        loggerFactory = new FlogLoggerFactory<>();
    }

    public ILoggerFactory getLoggerFactory() {
        return loggerFactory;
    }

    public String getLoggerFactoryClassStr() {
        return loggerFactoryClassStr;
    }
}